android/src/toga_android/fonts.py

```diff
@@ -22,13 +23,11 @@ def __init__(self, interface):
         self.interface = interface
 
     def get_size(self):
-        # Default system font size on Android is 14sp (sp = dp, but is separately
-        # scalable in user settings). For what it's worth, Toga's default is 12pt.
         if self.interface.size == SYSTEM_DEFAULT_FONT_SIZE:
-            font_size = 14
+            # Default system font size on Android is 14sp.
+            return TypedValue.COMPLEX_UNIT_SP, 14.0
         else:
-            font_size = self.interface.size
-        return float(font_size)
+            return TypedValue.COMPLEX_UNIT_PT, float(self.interface.size)
```

Comment:
I don't think the default font size is the same for all widgets, so it's probably better to cache the initial size, just as we do for text color.

Also, when I changed the code to use `COMPLEX_UNIT_PT`, I only ran the testbed. But the next day when I came to run an actual app, I noticed it made the text too big. This makes sense, because while 14 pt (0.19 inches) might be a good body text size for a desktop screen, it's too large for a mobile.

So to allow reasonable cross-platform portability, I think we'll have to map the Toga `font_size` to something smaller than a literal point on Android. 1 SP might be about right, but this is something that can only be determined visually. 

In any case, font sizes should always be measured in SP, otherwise the [system font size setting](https://support.google.com/accessibility/android/answer/11183305) will have no effect on it.

Reply:
Hrm - that poses a problem, because Toga specifies all fonts in points. That means we're going to have to find some way to interpret points as SP in a meaningful way that respects font size selection and device resolution.

Reply:
Making 1 Toga point equal 1 Android SP gives results which are pretty consistent with Cocoa, so I've reverted to that. And I've verified manually that it responds correctly to the system font size setting, but I think we'll have to do without an automated test of that for now.

It looks like most widgets use a default size of 14sp, except the TextInput which is about 18sp. So I've made it cache the default size in the same way as color, and applied that to all the TextView-based widgets by making them inherit from a common base class.